python/py-flatbuffers/Portfile

```diff
@@ -0,0 +1,36 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+PortGroup           python 1.0
+
+name                py-flatbuffers
+# version should be the same as devel/flatbuffers
+version             1.12.0
+license             Apache-2
+platforms           darwin linux
+maintainers         {gmail.com:herby.gillot @herbygillot} \
+                    openmaintainer
+
+description         Python runtime library for use with the Flatbuffers \
+                    serialization format.
+long_description    {*}${description}
+
+homepage            https://google.github.io/flatbuffers/
+
+python.versions     38
+
+if {${name} ne ${subport}} {
+
+    depends_build-append \
+                    port:flatbuffers \
+                    port:py${python.version}-setuptools
+
+    # Do not try to fetch or checksum anything as we will be using the
+    # Python flatbuffers source already present in the flatbuffers port
+    fetch           {}
+    checksum        {}
+
+    extract {
+        copy ${prefix}/share/flatbuffers/python ${worksrcpath}/
```

Comment:
This is unusual. I can't think of any other port that gets its source from another port. It has the potential for things to get out of sync: for example, if flatbuffers is updated to 1.13.0 and py-flatbuffers is not, then installing py-flatbuffers @1.12.0 could actually be installing version 1.13.0.

Why does flatbuffers install the python source to ${prefix}/share/flatbuffers/python? If it's just copying the code out of the source tarball, then I would recommend that flatbuffers stop doing that and py-flatbuffers can fetch the same tarball and get the source code out of it itself. Use `dist_subdir flatbuffers` so that we don't need to mirror (and the user doesn't need to download) the same distfile twice.

If on the other hand the flatbuffers port generates the python source code so there is no other option but for py-flatbuffers to use it, then I recommend making py-flatbuffers verify that its `${version}` matches the version of flatbuffers (e.g., using `${prefix}/bin/flatc --version`) before allowing the build to proceed.

Reply:
`dist_subdir` is a great idea.  I think the ideal would have been making `py-flatbuffers` a sub-port of `flatbuffers`, as I had mentioned here: https://github.com/macports/macports-ports/pull/8118#discussion_r471144049